src/zenml/post_execution/pipeline_run.py

```diff
@@ -140,6 +140,21 @@ def enable_cache(self) -> Optional[bool]:
 
         return self.model.pipeline_configuration.get(PARAM_ENABLE_CACHE)
 
+    @property
+    def enable_artifact_metadata(self) -> Optional[bool]:
```

Comment:
I think I already commented on this in teh past, i feel the phrasing just feels wrong, can't we rename all the new methods to show that they refer to a state `artifact_metadata_enabled` instead of making it sound like you can use this to set something to enabled

Reply:
For example the param in the step environment is called `step_enabled`

Reply:
`caching_enabled` in the step environment is the processed version of this setting though, i.e., it takes both the pipeline setting and step setting into account, whereas these `enable_...` properties merely reflect the setting that the user specified on the respective object. @schustmi and I discussed the other day whether we should add `caching_enabled` into the step config as well but decided against it since we thought having it together with `enable_cache` might be too confusing.

Reply:
IMO it's also not worth spending a lot of effort on the post exec class property naming, I'd rather think about how we can further abstract those away or potentially even deprecate the post exec classes entirely